packages/jira-adapter/src/client/client.ts

```diff
@@ -73,7 +74,7 @@ export default class JiraClient extends clientUtils.AdapterHTTPClient<
       // The http_client code catches the original error and transforms it such that it removes
       // the parsed information (like the status code), so we have to parse the string here in order
       // to realize what type of error was thrown
-      if (e.message.endsWith('Request failed with status code 404')) {
+      if (e.response?.status === 404) {
```

Comment:
nit: add a check that e is an HTTP error 
```suggestion
      if (e instanceof HTTPError && e.response.status === 404) {
```